src/lib/package-generator.ts

```diff
@@ -91,44 +91,86 @@ function createMetadataJSON(typing: TypingsData): string {
 	return JSON.stringify(typing, replacer, 4);
 }
 
-function createPackageJSON(typing: TypingsData, fileVersion: number, availableTypes: { [name: string]: TypingsData }): string {
-	const dependencies: { [name: string]: string } = {};
-	function addDependency(d: string) {
-		if (availableTypes.hasOwnProperty(d)) {
-			const type = availableTypes[d];
-			// In normal releases, we want to allow patch updates, so we use `foo.bar.*`.
-			// In a prerelease, we can only reference *exact* packages.
-			// See https://github.com/npm/node-semver#prerelease-tags
-			const patch = settings.prereleaseTag ?
-				`${Versions.getLastVersion(type).lastVersion}-${settings.prereleaseTag}` :
-				"*";
-			const semver = `${type.libraryMajorVersion}.${type.libraryMinorVersion}.${patch}`;
-			dependencies[fullPackageName(d)] = semver;
+function filePath(typing: TypingsData, fileName: string): string {
+	return path.join(typing.root, fileName);
+}
+
+async function createPackageJSON(typing: TypingsData, fileVersion: number, availableTypes: { [name: string]: TypingsData }): Promise<string> {
+	// typing may provide a partial `package.json` for us to complete
+	const pkgPath = filePath(typing, "package.json");
+	interface PartialPackageJson {
+		dependencies?: { [name: string]: string };
+		version?: string;
+		description: string;
+	}
+	let pkg: PartialPackageJson = typing.hasPackageJson ?
+		parseJson(await fsp.readFile(pkgPath, { encoding: "utf8" })) :
+		{};
+
+	const ignoredField = Object.keys(pkg).find(field => !["dependencies", "version", "description"].includes(field));
+	if (ignoredField) {
+		throw new Error(`Ignored field in ${pkgPath}: ${ignoredField}`);
+	}
+
+	let dependencies = pkg.dependencies;
+	if (!dependencies) {
+		dependencies = {};
+		function addDependency(d: string): void {
+			if (availableTypes.hasOwnProperty(d)) {
+				const type = availableTypes[d];
+				// In normal releases, we want to allow patch updates, so we use `foo.bar.*`.
+				// In a prerelease, we can only reference *exact* packages.
+				// See https://github.com/npm/node-semver#prerelease-tags
+				const patch = settings.prereleaseTag ?
+					`${Versions.getLastVersion(type).lastVersion}-${settings.prereleaseTag}` :
+					"*";
+				const semver = `${type.libraryMajorVersion}.${type.libraryMinorVersion}.${patch}`;
+				dependencies[fullPackageName(d)] = semver;
+			}
 		}
+		typing.moduleDependencies.forEach(addDependency);
+		typing.libraryDependencies.forEach(addDependency);
 	}
-	typing.moduleDependencies.forEach(addDependency);
-	typing.libraryDependencies.forEach(addDependency);
 
-	let version = `${typing.libraryMajorVersion}.${typing.libraryMinorVersion}.${fileVersion}`;
-	if (settings.prereleaseTag) {
-		version = `${version}-${settings.prereleaseTag}`;
+	let version = pkg.version;

```

Comment:
Let's block it for now and we can unblock it if we find a reason to need it (I can't think of one, since you can provide the version in the file header)